diff --git a/mlx/backend/cuda/worker.cpp b/mlx/backend/cuda/worker.cpp
--- a/mlx/backend/cuda/worker.cpp
+++ b/mlx/backend/cuda/worker.cpp
@@ -50,7 +50,7 @@ void Worker::commit(cudaStream_t stream) {
 void Worker::thread_fn() {
+  uint64_t current_batch = 0;
   while (!stop_) {
-    uint64_t current_batch = 0;
     Tasks tasks;
     {
       std::unique_lock<std::mutex> lk(mtx_);
       cond_.wait(lk, [this, &current_batch] {
